astropy/io/ascii/core.py

```diff
@@ -1466,7 +1472,13 @@ def _get_writer(Writer, fast_writer, **kwargs):
     if 'exclude_names' in kwargs:
         writer.exclude_names = kwargs['exclude_names']
     if 'fill_values' in kwargs:
-        writer.data.fill_values = kwargs['fill_values']
+        # Prepend user-specified values to the class default.
+        with suppress(TypeError, IndexError):
+            # Test if it looks like (match, replace_string, optional_colname),
+            # in which case make it a list
+            kwargs['fill_values'][1] + ''
+            kwargs['fill_values'] = [kwargs['fill_values']]
+        writer.data.fill_values = kwargs['fill_values'] + writer.data.fill_values
```

Comment:
Sorry. You are right, I was getting confused, please disregard my example. However, the point still stands.
If I had a custom writer class with the following replacements:
missing -> -999
nan -> -888
'abc'->'def'
and I call it with `fill_values = ('123', '456')` then before this PR I would only have the 123->456 replacement. After this PR I will have 123->456 *plus* the default replacements missin->-999, nan->-888, and 'abs'->'cde'. Right?

In other words, before this change, I could switch off the default fill values by passing in essentially an empty list (or non-sense replacements that never happen. Now, I would have ot know the default replacements. If I want to witch them off in the example above, I would have to do (pseudo code) `fill_values=[(missing, missing), (nan,nan), ('abc', 'abc')]`.
Usually, default replacements in a class make sense and you usually want to keep them even if the user passes in a fill_value specificer to replace some other type of values, but it's stiull a change in behaviour.